extension/llm/src/function/create_embedding.cpp

```diff
@@ -0,0 +1,75 @@
+#include "common/assert.h"
+#include "common/exception/connection.h"
+#include "function/function.h"
+#include "function/llm_functions.h"
+#include "function/scalar_function.h"
+#include "httplib.h"
+#include "json.hpp"
+
+using namespace kuzu::common;
+using namespace kuzu::binder;
+using namespace kuzu::function;
+using namespace kuzu::catalog;
+using namespace kuzu::processor;
+
+namespace kuzu {
+namespace llm_extension {
+
+static void execFunc(const std::vector<std::shared_ptr<common::ValueVector>>& parameters,
+    const std::vector<common::SelectionVector*>& /*parameterSelVectors*/,
+    common::ValueVector& result, common::SelectionVector* resultSelVector, void* /*dataPtr*/) {
+    // This iteration only supports using Ollama with nomic-embed-text.
+    // The user must install and have nomic-embed-text running at
+    // http://localhost::11434.
+    KU_ASSERT(parameters.size() == 1);
+    httplib::Client client("http://localhost:11434");
+    httplib::Headers headers = {{"Content-Type", "application/json"}};
+    result.resetAuxiliaryBuffer();
+    for (auto selectedPos = 0u; selectedPos < resultSelVector->getSelSize(); ++selectedPos) {
+        std::string text = parameters[0]->getValue<ku_string_t>(selectedPos).getAsString();
+        nlohmann::json payload = {{"model", "nomic-embed-text"}, {"prompt", text}};
+        auto res = client.Post("/api/embeddings", headers, payload.dump(), "application/json");
```

Comment:
Not related to this PR. But I wonder if in the future we can spend multiple strings in the http request to reduce the total number of requests made.